tools/clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -7853,6 +7851,8 @@ def err_hlsl_load_from_mesh_out_arrays: Error<
    "output arrays of a mesh shader can not be read from">;
 def err_hlsl_out_indices_array_incorrect_access: Error<
    "a vector in out indices array must be accessed as a whole">;
+def err_hlsl_unsupported_long_vector
+    : Error<"vectors of over 4 elements in %0 are not supported">;
```

Comment:
DXC has not historically done this well, but it would be better if our diagnostics followed Clang convention:
https://github.com/microsoft/DirectXShaderCompiler/blob/main/tools/clang/docs/InternalsManual.rst#the-format-string

Specifically, arguments that are just going to be constant strings are discouraged, instead you should use `%select`

Reply:
I don't know if you are requesting a change or validating the change I made before

Reply:
I'm requesting that you use `%select` to select a substring, instead of passing string arguments to `Diag`.

Reply:
I'll bend to the guidelines, but having `<< 0` instead of `<< "ConstantBuffers or TextureBuffers` doesn't seem more readable to me.